### ISSUE TITLE
New .htaccess change pushed with 2.4.2 breaks upgrading if magento installed in a subdirectory? 

### ISSUE THREAD

**Issue by iphigenie** [magento/magento2#32179]
I have just done a 2.4.2 upgrade on a test site, one that was set up with Magento2 installed in a subdirectory

I didnt see it documented in release notes (could have missed it) but the deployed .htaccess implements the "move to /pub" security recommendation. The .htaccess as deployed will not work if Magento is installed in a subfolder - it hardcoded the redirect as /pub/ 

The commit https://github.com/magento/magento2/commit/640cad53009b291334234ccd61ab79f256b43da2 deploys these changes

release notes and upgrade notes don't mention it, though installation notes does mention it.

This affects new installs too

**Comment by mrtuvn** [comment]
Actually it's not hard-coded as you think. Seem now magento already move root folder to pub

**Comment by iphigenie** [comment]
> Actually it's not hard-coded as you think. Seem now magento already move root folder to pub

Yes it is hard coded and this certainly should not have been rolled out without **mention in the release notes or the "incompatible changes"** 

This has appeared in the installation instructions - but maybe only a week ago, not prior to release - but it is not mentioned in the release notes and you think something that will demand major SEO and architecture changes for anyone who is using more than Magento should perhaps have been given a bit more visibility.

**Comment by mrtuvn** [comment]
Maybe @shiftedreality know this more than me

**Comment by shiftedreality** [comment]
I've provided an update on #32285. It has also a workaround from the reporter.

**Comment by iphigenie** [comment]
thanks @shiftedreality I see they went through several steps to "restore" the old way. 

Obviously this will break on any upgrade and need to be redone over and over, so I would rather find ways to make the "new way" compatible with pub being a subfolder, not the root, and an htaccess that protects other folders funneling to the root.

in summary making pub the root:
- not possible on every hosting option
- breaks magento installed in a subfolder
- breaks any apps installed in OTHER subfolders, say a cms - you now have to modify your magento .htaccess for these, instead of just having their configuration nicely separate. And every upgrade will break it
- seems likely it would also break a site using "static files on subdomain" - at least on my site they all use pub/... visibly in the paths? Though I haven't yet gotten to the point of reinstating that so perhaps it works

It's quite flabbergasting this was snuck in without clear documentation of the changes - also flabbergasting the installation instructions just changed to have "pub is root" as a prerequisite instead of having documentation of any other scenario

**Comment by shiftedreality** [comment]
@iphigenie I clearly understand your concern.

However, we do not guarantee undocumented installation ways to work. From the Magento side, we do not have any documentation or testing scenarios to ensure this is something we support.

Before release, we had pretty long public and partner availability phases to make sure it is being tested by the pre-release program participants from different angles.
The only case when this was caught was https://github.com/magento/magento2/pull/29176 where I described that neither of documented flows is broken.

I've provided more context on https://github.com/magento/magento2/issues/32285 as well.
